src/traits.d

```diff
@@ -398,6 +401,27 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
     alias isDeclX = isX!Declaration;
     alias isFuncX = isX!FuncDeclaration;
 
+    Expression isPkgX(bool function(Package) fp)
+    {
+        return isDsymX((Dsymbol sym) {
+                Package p;
+
+                if (Import imp = sym.isImport())
+                {
+                    if (!imp.pkg)
+                    {
+                        printf("unable to process forward-referenced import %s", imp.toChars());
```

Comment:
That might work. I thought the error() function needed a Loc object, though. I'm not sure how to get one in this context.